airbyte-integrations/bases/source-acceptance-test/unit_tests/test_json_schema_helper.py

```diff
@@ -208,3 +208,51 @@ def test_get_object_strucutre(object, pathes):
 )
 def test_get_expected_schema_structure(schema, pathes):
     assert get_expected_schema_structure(schema) == pathes
+
+
+@pytest.mark.parametrize(
+    "keys, num_paths, last_value",
+    [
+        (["option1"], 2, {"a_key": "a_value"}),
+        (["option2"], 1, ["value1", "value2"]),
+        (["nonexistent_key"], 0, None),
+        (["option1", "option2"], 3, ["value1", "value2"])
+    ],
+)
+def test_find_and_get_nodes(keys: List[Text], num_paths: int, last_value: Any):
+    schema = {
+        "title": "Key_inside_oneOf",
+        "description": "Tests that keys can be found inside lists of dicts",
+        "type": "object",
+        "properties": {
+            "credentials": {
+                "type": "object",
+                "oneOf": [
+                    {
+                        "type": "object",
+                        "properties": {
+                            "common": {"type": "string", "const": "option1", "default": "option1"},
+                            "option1": {"type": "string"},
+                        },
+                    },
+                    {
+                        "type": "object",
+                        "properties": {
+                            "common": {"type": "string", "const": "option2", "default": "option2"},
+                            "option1": {"a_key": "a_value"},
+                            "option2": ["value1", "value2"],
+                        },
+                    },
+                ],
+            }
+        },
+    }
+    schema_helper = JsonSchemaHelper(schema)
+    variant_paths = schema_helper.find_nodes(keys=keys)
+    assert len(variant_paths) == num_paths
+
+    if variant_paths:
+        values_at_nodes = []
+        for path in variant_paths:
+            values_at_nodes.append(schema_helper.get_node(path))
+        assert last_value in values_at_nodes
```

Comment:
It's hard to write lists of expected values as part of `parametrize` when the values themselves could also be lists (how do you distinguish between the two?), so i settled on just making sure that the right number are there, and the last expected one is also there, as a proxy for checking all expected paths were found

Reply:
I think that's reasonable for now!